streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroKafkaSpoutTranslator.java

```diff
@@ -104,17 +104,18 @@ public int read () throws IOException {
         }
 
         public int read (byte[] bytes, int off, int len) throws IOException {
-            Preconditions.checkNotNull(bytes, "Given byte array can not be null");
-            Preconditions.checkPositionIndexes(off, len, bytes.length);
+            if (bytes == null) {
+                throw new NullPointerException();
```

Comment:
nit: Good to have error message in the Exception or leave the current null check as earlier which takes care of that.
```
Preconditions.checkNotNull(bytes, "Given byte array can not be null");
```